experiment/graphql_issue_example.py

```diff
@@ -0,0 +1,154 @@
+            "    issues(first:100, after:$after) {\n"
+            "      nodes{\n"
+            "        number\n"
+            "        title\n"
+            "        state\n"
+            "        createdAt\n"
+            "        labels(first:100){\n"
+            "          nodes{\n"
+            "            name\n"
+            "          }\n"
+            "        }\n"
+            "        assignees(first:100){\n"
+            "          nodes{\n"
+            "            login\n"
+            "          }\n"
+            "        }\n"
+            "      }\n"
+            "      pageInfo{\n"
+            "        hasNextPage\n"
+            "        endCursor\n"
+            "      }\n"
+            "    }\n"
+            "  }\n"
+            "  rateLimit {\n"
+            "    limit\n"
+            "    cost\n"
+            "    remaining\n"
+            "    resetAt\n"
+            "  }\n"
+            "}\n")
+    variables = {"owner": owner, "name": name}
+    if after is not None:
+        variables["after"] = after
+    return do_github_graphql_request(query, token, variables)
+
+
+def get_all_issues(owner, name, token, issue_func, show_progress=False):
+    """gets all issues for a repo and applies issue_func to each.
+
+    Args:
+    owner: the GitHub repo owner as in github.com/kubernetes/test-infra ->
+            owner="kubernetes"
+    name: this GitHub repo name as in github.com.kubernetes/test-infra ->
+            name = "test-infra"
+    token: a GitHub API token string
+    issue_func: a function that takes one argument (the json of each issue)
+        this will be applied to each issue object returned by the GitHub API
+    show_progress: if True then print '.' for each request made
+
+    Raises:
+    IOError: an error occurred while getting the issues
+    """
+    response = get_issues(owner, name, token)
+    while True:
+        if show_progress:
+            print(".", end="")
+            sys.stdout.flush()
+        if response.status_code != 200:
+            raise IOError("failed to fetch issues for repo: %s/%s" % (owner, name))
+        response_json = response.json()
+        # NOTE: this will also contain the rate limit info if we need that later
+        # https://developer.github.com/v4/guides/resource-limitations/
+        data = response_json["data"]
+        issues = data["repository"]["issues"]["nodes"]
+        for entry in issues:
+            issue_func(entry)
+        page_info = data["repository"]["issues"]["pageInfo"]
+        if not page_info["hasNextPage"]:
+            break
+        response = get_issues(owner, name, token, page_info["endCursor"])
+
+
+def main():
+    token = sys.argv[-1]
+    org = "kubernetes"
+    repo = "test-infra"
+    print("getting issues for: %s/%s" % (org, repo))
+    # TODO: replace this with with something more useful?
+    def issue_func(issue):
+        print(issue)
+    get_all_issues(org, repo, token, issue_func)
```

Comment:
`show_progress=True`?

Reply:
It's just printing the issues now so showing progress seemed redundant. we could dump these to a CSV or something but they have nested json. I don't actually have a good use case for this script besides showing how to iterate the issues with graphql.